presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -485,12 +485,16 @@ protected RelationPlan visitValues(Values node, Void context)
                 for (int i = 0; i < items.size(); i++) {
                     Expression expression = items.get(i);
                     expression = ExpressionTreeRewriter.rewriteWith(new ParameterRewriter(analysis.getParameters(), analysis), expression);
+                    // RelationPlanner should not invoke evaluateConstantExpression, which in turn invokes ExpressionInterpreter.
+                    // This should happen in an optimizer.
                     Object constantValue = evaluateConstantExpression(expression, analysis.getCoercions(), metadata, session, analysis.getColumnReferences(), analysis.getParameters());
                     values.add(LiteralInterpreter.toExpression(constantValue, scope.getRelationType().getFieldByIndex(i).getType()));
                 }
             }
             else {
                 row = ExpressionTreeRewriter.rewriteWith(new ParameterRewriter(analysis.getParameters(), analysis), row);
+                // RelationPlanner should not invoke evaluateConstantExpression, which in turn invokes ExpressionInterpreter.
+                // This should happen in an optimizer.
```

Comment:
I thought that this is happening in optimizer now.

Reply:
See com.facebook.presto.sql.planner.iterative.rule.SimplifyExpressions#rewrite. I wonder why it needs to be here as well.